### PR TITLE
Teach --karg-delete option to ostree admin deploy

### PR DESCRIPTION
**Pull request by HuijingHei**
See https://github.com/ostreedev/ostree/issues/2586

### CODE REVIEW

**Review by travier — APPROVED** [review]

File: src/ostree/ot-admin-builtin-deploy.c [hunk 1]
```diff
@@ -90,6 +92,18 @@ ot_admin_builtin_deploy (int argc, char **argv, OstreeCommandInvocation *invocat
     return FALSE;
   }
 
+  if (opt_kernel_arg_none && opt_kernel_argv_delete)
+  {
```
Comment (cgwalters) [inline src/ostree/ot-admin-builtin-deploy.c:96]: GNU style has two spaces for indent here.  But I think we'll eventually do https://github.com/ostreedev/ostree/pull/2565 anyways; feel free to fix in a later patch though!

**Review by cgwalters — APPROVED** [review]
Thanks, nice and simple!

**Comment by bauen1** [comment]
Is it intentional, that `--karg-delete` can only be specified **once** ?

That is quite unlike `--karg-append` and makes it significantly less useful, additionally no warning (or rather error) is issued when it is specified multiple times.

**Comment by cgwalters** [comment]
I see no reason not to support multiple invocations of `--karg-delete` indeed.

**Pull request merged**